src/rtos/rtos.cpp

```diff
@@ -99,11 +99,31 @@ using namespace pros::c;
 
   Mutex::Mutex(void) : mutex(mutex_create(), mutex_delete) { }
 
+  Clock::time_point Clock::now() {
+    return Clock::time_point{Clock::duration{millis()}};
+  }
+
   bool Mutex::take(std::uint32_t timeout) {
     return mutex_take(mutex.get(), timeout);
   }
 
   bool Mutex::give(void) {
     return mutex_give(mutex.get());
   }
+
+  void Mutex::lock(void) {
+	  if (!take(TIMEOUT_MAX)) {
+		  throw std::runtime_error{"Cannot obtain lock!"};
+	  }
+  }
+
+  void Mutex::unlock(void) {
+	  give();
+  }
+
+  bool Mutex::try_lock(void) {
+	  try_lock_for(std::chrono::duration<long, std::ratio<1>>{1});
+	  try_lock_until(Clock::time_point{std::chrono::duration<long, std::ratio<1>>{1}});
+	  return take(0);

```

Comment:
Can you explain this logic? Why is the mutex trying to be taken 3 times?

Reply:
i have a feeling it's an attempt to artificially recreate the behaviour described in

> This function is allowed to fail spuriously and return false even if the mutex is not currently locked by any other thread.

which [someone explains](https://stackoverflow.com/a/48305054) probably really means one of two things:

1. the mutex is already locked elsewhere...
2. some platform-specific feature interrupts or prevents the locking attempt, and control is returned to the caller who can decide whether to retry.

in which case i think we really can just cut this down to `take(0)`... input appreciated

Reply:
Original source of the fail spuriously: https://en.cppreference.com/w/cpp/thread/mutex/try_lock

Another thing to note about this, is that the same source says that the function "Returns immediately.", so we could use that to justify not waiting for those attempts and just cutting it down to take(0);

Reply:
agreed. I think the default behaviour of a freeRTOS mutex matches the desired pretty closely anyway (i.e. if something prevents the locking attempt it will fail properly... probably)